lib/OpenAPI/Client/OpenAI.pm

```diff
@@ -108,172 +148,418 @@ Create a new OpenAI API client. The following options can be provided:
 The path to the OpenAPI specification file (YAML). Defaults to the
 "openai.yaml" file in the distribution's "share" directory.
 
+You can find the latest version of this file at
+L<https://github.com/openai/openai-openapi>.
+
 =back
 
-Additional options are passed to the parent class, OpenAPI::Client.
+Additional options are passed to the parent class, OpenAPI::Client, with the
+exception of the following extra options:
 
-=head2 Completions
+=over
 
-=head3 createCompletion
+=item * C<assistants>
 
-Creates a completion for the provided prompt and parameters.
+If set to a true value, this will allow access to the Assistants API.
 
-=head2 Chat Completions
+    my $client = OpenAPI::Client::OpenAI->new(
+        undef,    # use the default specification file
+        assistants => 1,    # enable the Assistants API
+    );
 
-=head3 createChatCompletion
+=back
 
-Creates a completion for the chat message.
+=head1 METHODS
 
-=head2 Edits
+For the parameters for each method, please consult the F<share/openapi.yaml>
+document, or the OpenAI API specification you passed to the constructor, if
+different.
 
-=head3 createEdit
+=head2 listAssistants
 
-Creates a new edit for the provided input, instruction, and parameters.
+Returns a list of assistants.
 
-=head2 Images
+    my $response = $client->listAssistants( { body => \%params });
 
-=head3 createImage
+=head2 createAssistant
 
-Creates an image given a prompt.
+Create an assistant with a model and instructions.
 
-=head3 createImageEdit
+    my $response = $client->createAssistant( { body => \%params });
 
-Creates an edited or extended image given an original image and a prompt.
+=head2 deleteAssistant
 
-=head3 createImageVariation
+Delete an assistant.
 
-Creates a variation of a given image.
+    my $response = $client->deleteAssistant( { body => \%params });
 
-=head2 Embeddings
+=head2 getAssistant
 
-=head3 createEmbedding
+Retrieves an assistant.
 
-Creates an embedding vector representing the input text.
+    my $response = $client->getAssistant( { body => \%params });
 
-=head2 Audio
+=head2 modifyAssistant
 
-=head3 createTranscription
+Modifies an assistant.
+
+    my $response = $client->modifyAssistant( { body => \%params });
+
+=head2 createSpeech
+
+Generates audio from the input text.
+
+    my $response = $client->createSpeech( { body => \%params });
+
+=head2 createTranscription
 
 Transcribes audio into the input language.
 
-=head3 createTranslation
+    my $response = $client->createTranscription( { body => \%params });
```

Comment:
it would be better if we had a example of this method (and any other file upload methods) that use the `file_upload` incantation that actually works (unless we update the generated endpoints to handle that).

This actually opens a question wherein we could have a preprocessor which wraps the `$params` in `body` for you; are people using this enough for us to care about backcompat?

Reply:
I used LWP in Intellexer::API for file uploads...

sub recognizeNeFileContent($self, $file, %params){

```perl
my $uri_obj= $self->_build_url(
        'recognizeNeFileContent?',
        %params
    );

    return $self->_react(
        $ua->post(
            $uri_obj,
            Content_Type => 'multipart/form-data',
            Content => [file => [$file,] ],
        )
    );
}
```